### ISSUE TITLE
It is not possible to do evaluation on some datasets with `lmms-eval`

### ISSUE THREAD

**Issue by ByUnal** [DeepLearnXMU/AVG-LLaVA#10]
Hi there,

I've completed all training step, and now I have final code to evaluate. However, when I try to evaluate my model on some datasets like `textvqa_test`, `docvqa_test`, or `vizwiz_vqa_test`, I'm having issues. The code doesn't throw an error, but there are problem in their target values so that the metrics are not measurable. I'll provide samples from evaluations of those datasets:

```
{
    "doc_id": 0,
    "target": "answer",
    "doc": {
      "image_id": "6abb8ddf5bd0ac6d",
      "question_id": 39602,
      "question": "what brand of phone?",
      "question_tokens": [
        "what",
        "brand",
        "of",
        "phone"
      ],
      "image_width": 551,
      "image_height": 1024,
      "flickr_original_url": "https://c6.staticflickr.com/4/3181/2679337312_93cc6c9558_o.jpg",
      "flickr_300k_url": "https://c6.staticflickr.com/4/3181/2679337312_93cc6c9558_o.jpg",
      "answers": [
        "",
        "",
        "",
        "",
        "",
        "",
        "",
        "",
        "",
        ""
      ],
      "image_classes": [
        "Computer keyboard",
        "Mobile phone"
      ],
      "set_name": "test",
      "ocr_tokens": [
        "NOKIA",
        "E71",
        "Operator",
        "We",
        "Har",
        "for",
        "Exchange",
        "(12)",
        "Amardalund",
        "Busanes",
        "report",
        "HymallE5)",
        "0",
        "11",
        "12.00",
        "0c-12:005alesweeting",
        "seeting",
        "WLA",
        "network",
        "foond",
        "54",
        "EX)",
        "Camera",
        "0",
        "(in]",
        "G",
        "8",
        "a",
        "A",
        "S",
        "D",
        "OJ"
      ]
    },
    "arguments": [
      [
        "What brand of phone?\nAnswer the question using a single word or phrase.",
        0,
        "textvqa_test",
        "test"
      ]
    ],
    "resps": [
      [
        "\n\nSamsung"
      ]
    ],
    "filtered_resps": [
      "\n\nSamsung"
    ],
    "exact_match": 0.0,
    "submission": {
      "question_id": 39602,
      "answer": "samsung"
    }
  }
```
From above, you can see that the `"target": "answer",`, thus no evaluation can be made. This is same for all samples.

Another example from `docvqa_test`:
```
{
    "doc_id": 0,
    "target": null,
    "doc": {
      "questionId": "57344",
      "question": "What is the dividend payout in 2012?",
      "question_types": null,
      "docId": 4720,
      "ucsf_document_id": "rnbx0223",
      "ucsf_document_page_no": "193",
      "answers": null,
      "data_split": "test"
    },
    "arguments": [
      [
        "What is the dividend payout in 2012?\nAnswer the question using a single word or phrase.",
        0,
        "docvqa_test",
        "test"
      ]
    ],
    "resps": [
      [
        "1.25"
      ]
    ],
    "filtered_resps": [
      "1.25"
    ],
    "anls": {
      "questionId": 57344,
      "answer": "1.25"
    },
    "submission": {
      "questionId": 57344,
      "answer": "1.25"
    }
  },
  {
    "doc_id": 1,
    "target": null,
    "doc": {
      "questionId": "16384",
      "question": "What is the name of the person in the CC field ?",
      "question_types": null,
      "docId": 5160,
      "ucsf_document_id": "lflm0081",
      "ucsf_document_page_no": "1",
      "answers": null,
      "data_split": "test"
    },
    "arguments": [
      [
        "What is the name of the person in the CC field ?\nAnswer the question using a single word or phrase.",
        1,
        "docvqa_test",
        "test"
      ]
    ],
    "resps": [
      [
        "John"
      ]
    ],
    "filtered_resps": [
      "John"
    ],
    "anls": {
      "questionId": 16384,
      "answer": "John"
    },
    "submission": {
      "questionId": 16384,
      "answer": "John"
    }
  },
```
Since the target value is `null` here, also it is not possible to do any evaluation here. 

There lots issue on these dataset like ai2d, vizwiz, etc. How can I fix that ? Any suggestion would be really helpful.
(AI2D is also really problematic, but I saw that you've evaluated your model on ai2d as far as I see from the paper. Could you show me how ?)

**Comment by Nonoor** [comment]
Thanks for the question!

For datasets like TextVQA-test, DocVQA-test, the ground-truth answers are not publicly released. Therefore, target will appear as "answer" or null, and local evaluation is not possible.

To get scores, you need to submit the prediction files to the official evaluation servers.
Our evaluation follows lmms-eval, and similar issues may be discussed here: https://github.com/EvolvingLMMs-Lab/lmms-eval/issues

Use validation splits if you want local metrics.

**Comment by ByUnal** [comment]
Hi again,

Thanks for the answer

I used `lmms-eval` for the evaluations, but I'm confused about the evaluation metrics used for VQA datasets. I have two main questions:

1. Is `exact_match` the standard metric for VQA datasets?
When I evaluate my model on textvqa_val, gqa and ai2d using lmms-eval, I observe that only exact_match is calculated as the metric. However:

The official TextVQA benchmark uses an Accuracy metric with soft matching (normalization + partial credit if answer matches ≥3 out of 10 human annotations)
Qwen-VL uses "VQA Score", not exact match

Questions:
- Did you also use exact_match for evaluating on TextVQA, GQA, and AI2D in AVG-LLaVA paper ?
- Is exact_match in lmms-eval equivalent to the "Accuracy" metric reported in AVG-LLaVA paper ?
- If not, how did you configure lmms-eval to use the proper evaluation metrics ?
- Is it possible to evaluate accuracy instead of exact_match by using lmms-eval ?


2. Issue with multiple-choice answer formatting (AI2D example)
I'm seeing 0.0 exact match scores even when my model provides correct answers. Here's an example from AI2D:
```json
{
    "doc_id": 3,
    "target": "C",
    "doc": {
        "question": "According to the given food chain, which is the smallest organism?",
        "options": ["krill", "whale", "Phytoplankton", "leopard"],
        "answer": "2"
    },
    "resps": [["C. Phytoplankton"]],
    "filtered_resps": ["C. Phytoplankton"],
    "exact_match": 0.0
}
```
The issue:

My model correctly answers: "C. Phytoplankton"
The ground truth is: "2" (which corresponds to option C, zero-indexed)
Result: exact_match = 0.0 because "C. Phytoplankton" ≠ "2"

My questions here would be that:
- How did you handle this mismatch between model output format and ground truth format?
- Did you use additional post-processing to extract just the letter "C" from "C. Phytoplankton"?
- Or does lmms-eval have a specific configuration for multiple-choice questions that I'm missing?

I'm waiting your answer ASAP. Thanks

**Comment by Nonoor** [comment]
Hi, thanks for the detailed questions.

**1. Evaluation metrics (TextVQA / GQA / AI2D)**
We directly used the default configurations provided by lmms-eval, which is widely adopted in the community. We also evaluated LLaVA-1.5 and LLaVA-NeXT using the same settings, and the results closely matched their official reports. Therefore, we did not further inspect or modify the internal metric implementations.

**2. No additional post-processing**
We did not apply extra processing to model outputs.
For AI2D specifically, the prompt in our repo:
https://github.com/XMUDeepLIT/AVG-LLaVA/blob/master/lmms-eval/lmms_eval/tasks/ai2d/ai2d.yaml explicitly instructs: “Answer with the option's letter from the given choices directly.”

In your example, the model outputs “C. Phytoplankton”, but the task expects only “C”. Because the instruction is not followed, exact_match = 0.0 is expected.

If you want to support different answer formats, you may need to modify the task configs under:
https://github.com/XMUDeepLIT/AVG-LLaVA/tree/master/lmms-eval/lmms_eval/tasks/ai2d
you can refer to the latest lmms-eval version (https://github.com/EvolvingLMMs-Lab/lmms-eval/tree/main).

Hope this helps!
